app/src/main/java/com/example/dungeontest/Settings.kt

```diff
@@ -227,9 +221,9 @@ fun TextInputField(tokenValue: MutableState<TextFieldValue>, labelText: String)
     val focusManager = LocalFocusManager.current
     val textFieldWidth = (maxLength * 8).dp
     OutlinedTextField(
-        value = tokenValue.value,
-        onValueChange = {
-            tokenValue.value = it
+        value = tokenValueInput.value?.let { it } ?: tokenValueState,
```

Comment:
```suggestion
        value = tokenValueInput.value ?: tokenValueState,
```
per ide, this should be equivalent